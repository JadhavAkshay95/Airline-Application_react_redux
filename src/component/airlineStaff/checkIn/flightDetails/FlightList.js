import React, { useEffect } from 'react'
import Flight from './Flight'
import * as FlightAction from './../../../../redux/actions/flightAction'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'

export const FlightList = ({
  flightList,
  loadFlight,
  currentUser,
  history
}) => {
  useEffect(() => {
    loadFlight()
  }, [])
  return (
    <>
      <Flight
        flightList={flightList}
        history={history}
        currentUser={currentUser}
      />
      <br></br> <br></br>
      <br></br>
    </>
  )
}

const mapStateToProps = state => {
  return {
    flightList: state.flightReducer,
    currentUser: Array.isArray(state.loginReducer) ? null : state.loginReducer
  }
}

const mapDispatchToProps = {
  loadFlight: FlightAction.loadFlight
}

export default connect(mapStateToProps, mapDispatchToProps)(FlightList)
FlightList.propTypes = {
  flightList: PropTypes.array,
  loadFlight: PropTypes.func,
  currentUser: PropTypes.object,
  history: PropTypes.object
}
