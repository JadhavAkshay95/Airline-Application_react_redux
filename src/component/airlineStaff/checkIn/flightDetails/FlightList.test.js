import React from 'react'
import { shallow } from 'enzyme'
import { FlightList } from './FlightList'

const renderFlight = args => {
  const defaultProps = {
    flightList: [],
    match: {},
    loadFlight: () => {}
  }

  const props = { ...defaultProps, ...args }
  return shallow(<FlightList {...props} />)
}

describe('Render flight component', () => {
  it('Render flight list', () => {
    const flightList = [
      {
        id: 'f1',
        flightCompany: {
          name: 'Air India',
          logo: 'airindia.jpg',
          type: 'Economy'
        },
        time: {
          boardingTime: '12.55 AM',
          deparatureTime: '3',
          duration: '2 hr',
          type: 'non-stop'
        },
        flightSource: 'Pune',
        flightDest: 'Bengluru',
        stops: [],
        facility: {
          checkinBaggage: '15 kg',
          paidMeal: 'yes'
        },
        totalSeat: 100,
        passengerId: [1, 2]
      }
    ]
    const wrapper = renderFlight({ flightList })
    expect(wrapper).toMatchSnapshot()
  })
})
