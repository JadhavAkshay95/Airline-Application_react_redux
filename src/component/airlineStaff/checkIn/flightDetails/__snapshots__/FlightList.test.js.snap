// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render flight component Render flight list 1`] = `ShallowWrapper {}`;
